frontend/src/components/SignupForm/SignupForm.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useContext } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import './SignupForm.css'
+import { UserContext } from '../../UserContext.js';
+
+const SignupForm = () => {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```

Comment:
Potential Advanced Feature Opportunity:
1. Some kind of email format validation, so that entries like `foo@bar` can't be used as email
2. Enforce pattern matching for password, e.g must have a symbol
3. Ensure all usernames are unique, e.g username has been taken, please choose another.

Reply:
To do: 1. & 2.